internal/apimeta/mapper.go

```diff
@@ -0,0 +1,81 @@
+package apimeta
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+)
+
+// ResourceMapper allows callers to map resources to kinds.
+type ResourceMapper interface {
+	// NamespacedKindFor returns the priority mapping GVK for a namespaced GR.
+	// If the GR doesn't exist, or isn't namespaced, returns an error.
+	NamespacedKindFor(gr schema.GroupResource) (schema.GroupVersionKind, error)
+}
+
```

Comment:
Generally speaking, the producer module shouldn't declare the interface type: https://github.com/golang/go/wiki/CodeReviewComments#interfaces

Right now, only `internal/hncconfig` uses this type so I think it's fine to move this interface to that package (and make it private). Maybe we can move it back here if a bunch of other modules start implementing it to avoid having to redefine it multiple times.

Reply:
Thx, I wasn't aware of this practice in Golang. Fixed.